astroid/brain/brain_typing.py

```diff
@@ -132,6 +148,44 @@ def infer_typing_typevar_or_newtype(node, context_itton=None):
     return node.infer(context=context_itton)
 
 
+def infer_typing_newtype(
+    node: nodes.Call, context_itton: typing.Optional[context.InferenceContext] = None
+) -> typing.Iterator[nodes.ClassDef]:
+    """Infer a typing.NewType(...) call"""
+    try:
+        func = next(node.func.infer(context=context_itton))
+    except (InferenceError, StopIteration) as exc:
+        raise UseInferenceDefault from exc
+
+    if func.qname() != "typing.NewType":
+        raise UseInferenceDefault
+    if len(node.args) != 2:
+        raise UseInferenceDefault
+
+    derived, base = node.args
+    derived_name = derived.as_string().strip("'")
+    base_name = base.as_string().strip("'")
+
+    new_node: ClassDef = extract_node(
+        TYPING_NEWTYPE_TEMPLATE.format(derived=derived_name, base=base_name)
+    )
+    new_node.parent = node.parent
+
+    # Base type arg is a normal reference, so no need to do special lookups
+    if not isinstance(base, nodes.Const):
+        new_node.bases = [base]
```

Comment:
Doesn't the `extract_node` already infer this? I'm not 100% proficient with astroid internals so this might not be the case, but this seems like it should.

Reply:
That was my initial thought (and implementation) as well. However, in that case the inference wasn't working, and any attrs were coming back as `Uninferable`.

If I had to take a guess, the issue is that the `extract_node()` above doesn't know the context of the surrounding module, and so the base class name isn't resolved properly.

Looking elsewhere in the code, the impression that I got was that constructing the `ClassDef` directly and/or using `ClassDef.postinit()` was more common. So if there is a way to make `extract_node()` behave as desired, I wasn't able to find it.

Reply:
Sorry for taking so long to respond to this.

Can we use `ClassDef.postinit` here? I have recently dabbled with the `NamedTuple` brain and also found that resolving base classes can be quite difficult, so I know the struggle now 😄

Reply:
Switched to `postinit`